### DISCUSSION TITLE
set is not updating the state

### DISCUSSION THREAD

**Discussion by ulisesantana** [pmndrs/zustand#1337] [Q&A]
If you wrap the result of `create` in a function ends up not updating the state. No matter how many times you push the button, never is gonna update.

```js
const useStore = () =>
  create((set) => ({
    count: 1,
    inc: () =>
      set((state) => {
        console.log("COUNT", state.count);
        return { count: state.count + 1 };
      })
  }));

function Counter() {
  const { count, inc } = useStore()();
  return (
    <div className="counter">
      <span>{count}</span>
      <button onClick={inc}>one up</button>
    </div>
  );
}
```

I was trying this for injecting dependencies to the store. However, this is getting tricky. Here you have [a codesandbox with a demo](https://codesandbox.io/s/hof-with-zustand-3r88k6?file=/src/App.js:164-502).

**Comment by dai-shi** [comment]
You don't want to create a store in the render function. If that's your intention useRef should help keeping the store stable.
